examples/tests/valid/dynamo.ts

```diff
@@ -11,15 +16,15 @@ export async function _putItem(tableName, item) {
   const response = await client.send(command);
   console.log(response);
   return;
-}
+};
 
-export async function _getItem(tableName, key) {
+export const _getItem: extern["_getItem"] = async (tableName, key) => {
```

Comment:
Isn't there some typescript incantation that will apply this automatically to all extern declarations in this file?

Reply:
Unfortunetly not https://github.com/microsoft/TypeScript/issues/38511.

The closest would be doing a default export with the whole module:

```ts
import type extern from "./blah.extern";

export default {
  method1() {
  }
  method2() {
  }
} satisfies extern
```

This does work now, however it adds a runtime cost to export it this way and it actually breaks esbuild's bundling in some of our examples.

---

Some magic that should be possible is to make `extern` an implicit global type for a given file. That way you can avoid the `import type extern ...`. I had trouble getting it working so I figured I'd leave it as a future dx improvement.

Reply:
Cool, thanks for the info